### PR TITLE
Fix out of bounds segfault

### PR DESCRIPTION
**Pull request by GeorgWa**
This should capture out-of bounds based sefaults like https://github.com/MannLabs/alphadia/issues/260

### CODE REVIEW

File: alphadia/plexscoring.py [hunk 1]
```diff
@@ -1895,6 +1895,9 @@ def __call__(
         )
         thread_count = 1 if debug else thread_count
 
+        if debug:
```
Comment (mschwoer) [inline alphadia/plexscoring.py:1898]: could you gather this and the other two `if debug`s in l. 1893 and 1896 in one block?

File: alphadia/features.py [hunk 2]
```diff
@@ -362,16 +373,76 @@ def center_envelope_1d(x):
...
+    ----------
+    x: np.ndarray
+        Array of shape (a, b) where a is the number of arrays and b is the length of each array.
+        It is mandatory that dimension b is even.
+
+    """
+    center_index_right = x.shape[1] // 2
+    center_index_left = center_index_right - 1
+
+    for a0 in range(x.shape[0]):
```
Comment (mschwoer) [inline alphadia/features.py:403]: why `a0`? just `a` would be less noisy

**Review by mschwoer — APPROVED** [review]
LGTM

File: alphadia/features.py (hunk 2 above)
Comment (GeorgWa) [reply alphadia/features.py:403]: that's a pattern I reuse a lot across numba function for iterating axes. I would keep it as long as I don't remove it completely from the codebase.

**Pull request merged**